onboarding/source/amb_light_service.c

```diff
@@ -6,34 +6,80 @@
 
 /* USER CODE BEGIN */
 // Include any additional headers and global variables here
+#include <FreeRTOS.h>
+#include <os_task.h>
+#include <os_queue.h>
+
+static TaskHandle_t lightServiceTaskHandle = NULL;
+static QueueHandle_t lightServiceQueueHandle = NULL;
 
 /* USER CODE END */
 
 /**
  * @brief Task that waits for MEASURE_LIGHT event and then prints the ambient light value to the serial port.
  * @param pvParameters Task parameters
  */
-static void lightServiceTask(void * pvParameters);
+static void lightServiceTask(void *pvParameters);
 
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned;
+
+    if (lightServiceTaskHandle == NULL) {
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                lightServiceTaskHandle);
+    }
+
+    if (lightServiceQueueHandle == NULL) {
+        lightServiceQueueHandle = xQueueCreate(LIGHT_SERVICE_QUEUE_LENGTH,
+                                               LIGHT_SERVICE_QUEUE_ITEM_SIZE);
+    }
 
     /* USER CODE END */
-    return 1;
+    return 0;
```

Comment:
For function returns, we generally use 0 to represent a failure and 1/non-zero to represent success